### ISSUE TITLE
Confusion about paper “HydrasandIPFS:adecentralisedplaygroundformalware”

### ISSUE THREAD

**Issue by Sh4d0wBlade** [kpatsakis/RIGA#1]
Are you the author of this paper？

**Comment by Sh4d0wBlade** [comment]
Guess it is your paper
There is a formula in the paper like this "f (v i ) = h i , ∀i ∈ {1,2,...,k}" , hi is the hash address of a file exists, vi is the value of the counter and f is the PRNG to generate ipfs Qmxxxx addresses acording to your paper,but how to generate a hash address without knowing the contents of a object in IPFS? I mean,how to get the actual value of vi? This REALLY confuses me.

**Comment by kpatsakis** [comment]
Yes, I'm one of the authors. The idea is that you have precomputed the hashes and that you haven't uploaded them. This way, no one can know beforehand whether and if you will use any of these addresses.
